instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/internal/OpenTelemetryStatement.java

```diff
@@ -310,16 +320,11 @@ protected <T, E extends Exception> T wrapCall(String sql, ThrowingSupplier<T, E>
     return result;
   }
 
-  private String buildSqlForBatch() {
-    StringBuilder sqlBuilder = new StringBuilder();
-    if (query != null) {
-      sqlBuilder.append(query);
-    }
-
-    for (String batchCommand : batchCommands) {
-      sqlBuilder.append(batchCommand);
-    }
-
-    return sqlBuilder.toString();
+  protected <T, E extends Exception> T wrapBatchCall(ThrowingSupplier<T, E> callable) throws E {
+    DbRequest request =
+        this instanceof PreparedStatement
+            ? DbRequest.create(dbInfo, query, batchSize)
+            : DbRequest.create(dbInfo, batchCommands, batchSize);
+    return wrapCall(request, callable);
```

Comment:
does it help to override executeBatch() in OpenTelemetryPreparedStatement instead?

Reply:
implemented